### PR TITLE
Add pitch and roll rotation for skymap and skyboxes.

### LINKED ISSUE #1067: Create pitch/roll control for skymaps
Currently, the only rotation control for skymaps is _yaw_. A pitch/roll control would be useful for some skymaps. For example, the Milky Way Panorama by ESO is a full 360x180 skymap. The skymap can be rotated on the yaw axis, but the Milky Way will always be on the horizon. In real life, that is not the case; it runs up and down through the sky. If a pitch/roll control existed, the skymap could be positioned in any way desired.

### PR DESCRIPTION
**Pull request by leMaik**
Fixes #1067

### CODE REVIEW

File: chunky/src/java/se/llbit/chunky/renderer/scene/Sky.java [hunk 1]
```diff
@@ -476,18 +474,49 @@ private void addSunColor(Ray ray) {
...
 
   /**
-   * Set the polar offset of the skymap.
+   * Set the yaw rotation of the skymap.
+   * @deprecated Use {@link #setYaw(double)} instead.
    */
-  public void setRotation(double value) {
-    rotation = value;
-    scene.refresh();
+  public void setRotation(double yaw) {
```
Comment (ThatRedox) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Sky.java:480]:
```suggestion
  @Deprecated
  public void setRotation(double yaw) {
```

File: chunky/src/java/se/llbit/chunky/renderer/scene/Sky.java [hunk 2]
```diff
@@ -476,18 +474,49 @@ private void addSunColor(Ray ray) {
...
+    this.setYaw(yaw);
   }
 
   /**
-   * @return The polar offset of the skymap
+   * @return The yaw rotation of the skymap
+   * @deprecated Use {@link #getYaw()} instead.
    */
   public double getRotation() {
```
Comment (ThatRedox) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Sky.java:488]:
```suggestion
  @Deprecated
  public double getRotation() {
```

**Review by ThatRedox — CHANGES_REQUESTED** [review]
Extra `Vector3` allocation could impact performance; needs testing. Apart from that, looks good 👍

**Comment by jackjt8** [comment]
| 32 SPP                | memory  | sps    | render time |
|-----------------------|---------|--------|-------------|
| SNAPSHOT.196.ge9baf67 | 12245.7 | 290576 | 105         |
| PR.1473.203.g3163c56  | 10305.8 | 273550 | 111         |
|                       |         |        |             |
| 128 SPP               |         |        |             |
| SNAPSHOT.196.ge9baf67 | 19134.7 | 304801 | 403         |
| PR.1473.203.g3163c56  | 9846.9  | 314114 | 391         |

Looks like this PR doesn't impact performance... and I get some pretty cool results

**Comment by leMaik** [comment]
I updated the implementation so that it doesn't use intermediate `Vector3` instances anymore.

**Review by ThatRedox — APPROVED** [review]

**Pull request merged**